core/common/src/main/java/alluxio/retry/RetryUtils.java

```diff
@@ -46,6 +46,34 @@ public static void retry(String action, RunnableThrowsIOException f, RetryPolicy
     throw e;
   }
 
+  /**
+   * Retries the given method until it doesn't throw an IO exception or the retry policy expires. If
+   * the retry policy expires, the last exception generated will be rethrown.
+   *
+   * @param <T> type of the return value
+   * @param action a description of the action that fits the phrase "Failed to ${action}"
+   * @param f the function to retry
+   * @param policy the retry policy to use
+   * @return T return a value
+   */
+  public static <T> T retryFunction(String action, RunnableFunctionThrowsException<T> f,
```

Comment:
Can we not just overload the `retry` name?

Also, can we write a unittest for this method?